pink/barriers/position_barrier.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Stéphane Caron
+
+"""Frame position barrier."""
+
+from typing import List, Optional, Union
+
+import numpy as np
+
+from ..configuration import Configuration
+from .barrier import Barrier
+from .exceptions import NoPositionLimitProvided
+
+
+class PositionBarrier(Barrier):
+    r"""A position-based barrier.
+
+    Defines a barrier function based on the position of a
+    specified frame in the world coordinate system. It allows for the
+    specification of minimum and maximum position bounds along selected axes.
+
+    Attributes:
+        frame: Name of the frame to monitor.
+        indices: Indices of the position components to consider.
+        p_min: Minimum position bounds.
+        p_max: Maximum position bounds.
+    """
+
+    frame: str
+    indices: List[int]
+    p_min: Optional[np.ndarray]
+    p_max: Optional[np.ndarray]
+
+    def __init__(
+        self,
+        frame: str,
+        indices: Optional[List[int]] = None,
+        p_min: Optional[np.ndarray] = None,
+        p_max: Optional[np.ndarray] = None,
+        gain: Union[float, np.ndarray] = 1.0,
+        r: float = 0.0,
+    ):
+        """Initialize a position barrier.
+
+        Args:
+            frame (str): name of the frame to monitor.
+            indices (Optional[List[int]], optional): indices with limits.
```

Comment:
```suggestion
            indices: indices with limits.
```
No need to repeat type annotations here, Sphinx will carry them over from type hints above.